lib/s3_meta_sync/syncer.rb

```diff
@@ -83,6 +87,13 @@ def download(source, destination)
       end
     end
 
+    # Sometimes SIGTERM causes Dir.mktmpdir to not properly clear the temp folder
+    def clear_old_temp_folders
+      path = File.join(Dir.tmpdir, TEMP_FOLDER_PREFIX + '*')
+      total = Dir.glob(path).map { |dir| FileUtils.rm_rf(dir) }.count
```

Comment:
needs to do some age checking or multiple parallel (maybe parallel projects / downloads) runs will delete each others folders